core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -307,23 +309,31 @@ public void removeWorkflow(String workflowId, boolean archiveWorkflow) throws Ex
 	}
 
 	public SearchResult<WorkflowSummary> search(String query, String freeText, int start, int size, List<String> sortOptions) {
-		
+
 		SearchResult<String> result = indexer.searchWorkflows(query, freeText, start, size, sortOptions);
 		List<WorkflowSummary> workflows = result.getResults().stream().parallel().map(workflowId -> {
 			try {
-				
+
 				WorkflowSummary summary = new WorkflowSummary(executionDAO.getWorkflow(workflowId, false));
 				return summary;
-				
+
 			} catch(Exception e) {
 				logger.error(e.getMessage(), e);
 				return null;
 			}
 		}).filter(Objects::nonNull).collect(Collectors.toList());
-		int missing = result.getResults().size() - workflows.size();
+
+		//Search again to filter out some workflows that might have been in flux state before.
+		//It's possible that during the first search ES had some workflows that were already completed in dynamo but
+		//still had a 'RUNNING' state in ES
+		SearchResult<String> repeatedResult = indexer.searchWorkflows(query, freeText, start, size, sortOptions);
```

Comment:
I don't think we need to search again, we just need to filter List<WorkflowSummary> workflows by RUNNING status before return the list back.

Reply:
This search method is generic, meaning that the user might be searching by any status and other fields, using query or Lucene-based text search, so it's hard to extract exact search parameters to compare them with the retrieved workflow objects

Reply:
ah, that makes sense, didn't realize the method takes in a arbitrary query string.